Installation issue: likwid

### Steps to reproduce the issue

```console
$ sudo -i spack install likwid %gcc@9.1.0
...
==> 112438: Installing likwid
==> Fetching https://github.com/RRZE-HPC/likwid/archive/v5.0.1.tar.gz
######################################################################################################################################################### 100.0%     -=O=- #      #       #         #                                                                                                                          
==> Staging archive: /tmp/root/spack-stage/spack-stage-likwid-5.0.1-5p5zpttnzisogd6c274j5cezlbaf6uwc/v5.0.1.tar.gz
==> Created stage in /tmp/root/spack-stage/spack-stage-likwid-5.0.1-5p5zpttnzisogd6c274j5cezlbaf6uwc
==> Ran patch() for likwid
==> 112438: likwid: Building likwid [Package]
==> 112438: likwid: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

3 errors found in build log:
     119    ===>  COMPILE C GCCPOWER/bstrlib_helper.o
     120    ===>  COMPILE C GCCPOWER/bstrlib.o
     121    ===>  COMPILE C GCCPOWER/bench.o
     122    ===>  GENERATE BENCHMARKS
     123    ===>  ASSEMBLE  GCCPOWER/stream_triad_vsx4.o
     124    ./GCCPOWER/stream_triad_vsx4.s: Assembler messages:
  >> 125    ./GCCPOWER/stream_triad_vsx4.s:76: Error: unrecognized opcode: `jl'
     126    Makefile:129: recipe for target 'GCCPOWER/stream_triad_vsx4.o' failed
  >> 127    make[1]: *** [GCCPOWER/stream_triad_vsx4.o] Error 1
     128    Makefile:256: recipe for target 'bench/likwid-bench' failed
  >> 129    make: *** [bench/likwid-bench] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-likwid-5.0.1-5p5zpttnzisogd6c274j5cezlbaf6uwc/spack-build-out.txt

==> Error: Failed to install likwid due to ChildError: ProcessError: Command exited with status 2:
    'make' '-j16'
3 errors found in build log:
     119    ===>  COMPILE C GCCPOWER/bstrlib_helper.o
     120    ===>  COMPILE C GCCPOWER/bstrlib.o
     121    ===>  COMPILE C GCCPOWER/bench.o
     122    ===>  GENERATE BENCHMARKS
     123    ===>  ASSEMBLE  GCCPOWER/stream_triad_vsx4.o
     124    ./GCCPOWER/stream_triad_vsx4.s: Assembler messages:
  >> 125    ./GCCPOWER/stream_triad_vsx4.s:76: Error: unrecognized opcode: `jl'
     126    Makefile:129: recipe for target 'GCCPOWER/stream_triad_vsx4.o' failed
  >> 127    make[1]: *** [GCCPOWER/stream_triad_vsx4.o] Error 1
     128    Makefile:256: recipe for target 'bench/likwid-bench' failed
  >> 129    make: *** [bench/likwid-bench] Error 2

...
```

### Information on your system

* **Spack:** 0.14.2-1047-ff6a0fc7b
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-power8le

### Additional information

<!-- Please upload the following files. They should be present in the stage directory of the failing build. Also upload any config.log or similar file if one exists. -->
* [spack-build-out.txt](https://github.com/spack/spack/files/4673552/spack-build-out.txt)
* [spack-build-env.txt](https://github.com/spack/spack/files/4673553/spack-build-env.txt)

@TomTheBear 

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/likwid/package.py
+++ b/var/spack/repos/builtin/packages/likwid/package.py
@@ -29,6 +29,7 @@ class Likwid(Package):
     version('4.3.1', sha256='4b40a96717da54514274d166f9b71928545468091c939c1d74109733279eaeb1')
     version('4.3.0', sha256='86fc5f82c80fcff1a643394627839ec79f1ca2bcfad30000eb7018da592588b4')
 
+    patch('https://github.com/RRZE-HPC/likwid/commit/e0332ace8fe8ca7dcd4b4477a25e37944f173a5c.patch', sha256='c3b8f939a46b425665577ce764d4fba080a23cab5999c53db71655fd54d7e0b1', when='@5.0.1')
     patch('https://github.com/RRZE-HPC/likwid/commit/d2d0ef333b5e0997d7c80fc6ac1a473b5e47d084.patch', sha256='636cbf40669261fdb36379d67253be2b731cfa7b6d610d232767d72fbdf08bc0', when='@4.3.4')
 
     # NOTE: There is no way to use an externally provided hwloc with Likwid.

